### PR TITLE
Add intervention & restructure `simulate_tree_from_pop()`

### PR DESCRIPTION
**Pull request by jamesmbaazam**
This PR introduces two key changes:

- It partly addresses #3 by adding a feature to scale R0 through the mean of `offspring_dist`. The PR introduces the `intvn_scale_r0()` function for achieving this. The function only works for poison and negative binomial offspring distributions because their mean represents $R_0$. An error is thrown when `r0_reduction > 0` but `offspring_dist` is neither `pois` nor `nbinom`.  

- In the process of including this feature, it became necessary to restructure the `simulate_tree_from_pop()` function. It now uses `...` to pass extra arguments to `pois` and `nbinom`. The `offspring_disp` and `offspring_mean` arguments have been removed. This was not an issue previously but became necessary to streamline how `intvn_scale_r0()` works.

### CODE REVIEW

File: R/intervention.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#' @return List of the offspring distribution parameter(s) with the mean
+#' scaled by \code{1 - intvn_scale_r0}.
+#' @details
+#' `intvn_scale_r0()` scales the mean of the offspring distribution
+#' by \eqn{1 - r0\_reduction} so that the new mean is given as:
+#' \deqn{(1 - r0\_reduction) \times R_0,} where \eqn{R_0} is the
+#' mean of the poisson and negative binomial distribution.
+#'
+#' @author James M. Azam
+#' @export
```
Comment (sbfnk) [inline R/intervention.R:26]: does this function really need to be user-facing?

File: R/intervention.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+intvn_scale_r0 <- function(r0_reduction, offspring_dist, pars_list) {
```
Comment (sbfnk) [inline R/intervention.R:27]:
```suggestion
reduce_mean <- function(r0_reduction, offspring_dist, pars_list) {
```
Just a suggestion - I would go with a simpler name that reflects what the function does, not how we interpret it. Similarly could call the argument `mean_reduction` or similar. But it depends on how you envision this to work for other interventions, and if this reflects some kind of common interface/naming scheme.

File: R/intervention.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#' no intervention impact and `r0_reduction` = 1 implies full impact.
+#' @param pars_list Parameter(s) for poisson or negative binomial offspring
+#' distribution.
+#' @return List of the offspring distribution parameter(s) with the mean
+#' scaled by \code{1 - intvn_scale_r0}.
+#' @details
+#' `intvn_scale_r0()` scales the mean of the offspring distribution
+#' by \eqn{1 - r0\_reduction} so that the new mean is given as:
+#' \deqn{(1 - r0\_reduction) \times R_0,} where \eqn{R_0} is the
+#' mean of the poisson and negative binomial distribution.
```
Comment (sbfnk) [inline R/intervention.R:23]: Perhaps mention that this corresponds to the population-level reduction of R in Lloyd-Smith et al., as opposed to an individual-level reduction (X% of people have R0 set to 0). Eventually we might also want to implement the individual-level reduction, even though that is probably more tricky.

**Review by sbfnk — APPROVED** [review]
Nice work!

File: R/intervention.R (hunk 1 above)
Comment (jamesmbaazam) [reply R/intervention.R:26]: I'll unexpose it.

File: R/intervention.R (hunk 2 above)
Comment (jamesmbaazam) [reply R/intervention.R:27]: The plan is to use autocomplete for the intervention functions, hence, the prefix "intvn_". I thought "scale_R0" reads more naturally to the user. I was planning on calling the offspring truncation intervention "intvn_truncate_offspring".

File: R/intervention.R (hunk 3 above)
Comment (jamesmbaazam) [reply R/intervention.R:23]: > Perhaps mention that this corresponds to the population-level reduction of R in Lloyd-Smith et al., as opposed to an individual-level reduction (X% of people have R0 set to 0).

Oh, good point. I'll add that. 

> Eventually we might also want to implement the individual-level reduction, even though that is probably more tricky.

I am brainstorming in #3 how to go about that to capture generic use cases.

File: R/intervention.R (hunk 2 above)
Comment (jamesmbaazam) [reply R/intervention.R]: I have now renamed the function to `intvn_reduce_mean()` and the argument to `intvn_mean_reduction`.

**Pull request merged**